pybamm/models/base_model.py

```diff
@@ -382,13 +382,25 @@ def set_initial_conditions_from(self, solution, inplace=True):
         else:
             model = self.new_copy()
 
+        if isinstance(solution, pybamm.Solution):
+            solution = solution.last_state
+        else:
+            solution = pybamm.FuzzyDict(solution)
         for var, equation in model.initial_conditions.items():
             if isinstance(var, pybamm.Variable):
-                final_state = solution[var.name]
+                try:
+                    final_state = solution[var.name]
+                except KeyError as e:
+                    raise pybamm.ModelError(
+                        "To update a model from a solution, each variable in "
+                        "model.initial_conditions must appear in the solution with "
+                        "the same key as the variable name. In the solution provided, "
+                        f"{e.args[0]}"
+                    )
                 if isinstance(solution, pybamm.Solution):
                     final_state = final_state.data
                 if final_state.ndim == 1:
-                    final_state_eval = np.array([final_state[-1]])
+                    final_state_eval = final_state[-1:]
```

Comment:
Probably a stupid question, but is this colon necessary?

Reply:
Yeah, just discovered this in python, it returns an array (which is needed) instead of a scalar
```python3
np.array([1,2,3])[-1] # --> 3
np.array([1,2,3])[-1:] # --> np.array([3])
```